src/test/java/com/wikia/webdriver/testcases/seotests/HtmlTitleTests.java

```diff
@@ -13,218 +13,297 @@
             "À_propos",
             "À propos - Wikia",
+            VARIANT_ALL,
+        },
+        {
+            TEST_WIKI_CURATED_CONTENT,
+            "wiki/Main_Page",
+            "Wookieepedia - Wikia",
+            VARIANT_ONLY_MOBILE,
+        },
+        {
+            TEST_WIKI_CURATED_CONTENT,
+            "wiki/Droid_starfighter",
+            "Droid starfighter - Wookieepedia - Wikia",
+            VARIANT_ONLY_MOBILE,
+        },
+        {
+            TEST_WIKI_CURATED_CONTENT,
+            "main/category/Future_films",
+            // SEO-198: Future films - Wookieepedia - Wikia
+            "http://starwars.wikia.com/main/category/Future_films",
+            // XW-868
+            VARIANT_ONLY_ANONYMOUS_MOBILE,
+        },
+        {
+            TEST_WIKI_CURATED_CONTENT,
+            "main/section/Films",
+            // SEO-198: "Films - Wookieepedia - Wikia",
+            "http://starwars.wikia.com/main/section/Films",
+            // XW-868
+            VARIANT_ONLY_ANONYMOUS_MOBILE,
+        },
+        {
+            TEST_WIKI_DISCUSSION,
+            "d/f/3035/latest",
+            "Fallout Wiki - Wikia",
+            VARIANT_ONLY_MOBILE,
         },
+        {
+            TEST_WIKI_DISCUSSION,
+            "d/p/2594243417559008375",
+            "Fallout Wiki - Wikia",
+            VARIANT_ONLY_MOBILE,
+        },
+        {
+            TEST_WIKI_CHAT,
+            "wiki/Special:Chat",
+            // Starting from January 2016: "Chat - Sktest123 Wiki - Wikia",
+            "Sktest123 Wiki: Welcome to the Sktest123 Wiki chat - Wikia",
+            VARIANT_ONLY_LOGGED_IN_DESKTOP,
+        }
     };
   }
 
-    @DataProvider
-    private Object[][] dataHtmlTitleMercuryTest() {
-        return new Object[][]{
-            {
-                TEST_WIKI_CURATED_CONTENT,
-                "wiki/Main_Page",
-                "Wookieepedia, the Star Wars Wiki - Wikia",
-            },
-            {
-                TEST_WIKI_CURATED_CONTENT,
-                "wiki/Droid_starfighter",
-                "Droid starfighter - Wookieepedia, the Star Wars Wiki - Wikia",
-            },
-            {
-                TEST_WIKI_CURATED_CONTENT,
-                "main/category/Future_films",
-                "Future films - Wookieepedia, the Star Wars Wiki - Wikia",
-            },
-            {
-                TEST_WIKI_CURATED_CONTENT,
-                "main/section/Films",
-                "Films - Wookieepedia, the Star Wars Wiki - Wikia",
-            },
-            {
-                TEST_WIKI_DISCUSSION,
-                "d/f/3035/latest",
-                "The Fallout wiki - Fallout: New Vegas and more - Wikia",
-            },
-            {
-                TEST_WIKI_DISCUSSION,
-                "d/p/2594243417559008375",
-                "The Fallout wiki - Fallout: New Vegas and more - Wikia",
-            },
-        };
+  private boolean shouldSkipTestCase(int variant, boolean loggedIn) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "shouldSkipTestCase" is 15 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)